torch/utils/_sympy/value_ranges.py

```diff
@@ -175,6 +172,91 @@ def index_expr(self, index, dtype):
         assert isinstance(index, ValueRanges)
         return index
 
+    @staticmethod
+    def or_(a, b):
+        a = ValueRanges.wrap(a)
+        b = ValueRanges.wrap(b)
+        if a.lower or b.lower:
+            return ValueRanges.wrap(sympy.true)
+        elif a.lower == a.upper and b.lower == b.upper:
+            return ValueRanges.wrap(sympy.Or(a.lower, b.lower))
+        else:
+            return ValueRanges(sympy.false, sympy.true)
+
+    @staticmethod
+    def and_(a, b):
+        a = ValueRanges.wrap(a)
+        b = ValueRanges.wrap(b)
+        if not a.upper or not b.upper:
+            return ValueRanges.wrap(sympy.false)
+        elif a.lower == a.upper and b.lower == b.upper:
+            return ValueRanges.wrap(sympy.And(a.lower, b.lower))
+        else:
+            return ValueRanges(sympy.false, sympy.true)
+
+    @staticmethod
+    def eq(a, b):
+        a = ValueRanges.wrap(a)
+        b = ValueRanges.wrap(b)
+        if a.lower == a.upper and b.lower == b.upper and a.lower == b.lower:
```

Comment:
`ValueRanges.unknown() == ValueRanges.unknown() => True`. All of these functions are not handling infs correctly.

Reply:
Just to reiterate, on the code on this PR:

```
>>> ValueRangeAnalysis.eq(ValueRanges.unknown(), ValueRanges.unknown())
ValueRanges(lower=False, upper=True)
```

which is correct.